views/v_ksm_data_mart.sql

```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
-       LEFT JOIN KLC ON KLC.GIFT_CLUB_ID_NUMBER = deg.ID_NUMBER
-       LEFT JOIN anonymous ON anonymous.id_number = deg.ID_NUMBER
-       --- Dropping the anonymous donors
-       where anonymous.id_number is null
+Where GIFT_CLUB_CODE = 'LKM'),
+
+
+KLC_Give_Ind As (select KLC.GIFT_CLUB_ID_NUMBER,
+Max(Case When cal.curr_fy = Club_END_DATE Then 'Yes' Else NULL End) as KSM_donor_cfy,
+Max(Case When cal.curr_fy = Club_END_DATE + 1 Then 'Yes' Else NULL End) as KSM_donor_pfy1
+from KLC
+cross join rpt_pbh634.v_current_calendar cal
+group BY KLC.GIFT_CLUB_ID_NUMBER),
+
+--- Count Total KLC Years 
+
+KLC_Count As (Select distinct KLC.GIFT_CLUB_ID_NUMBER,
+Count (distinct Club_END_DATE) as klc_fy_count
+from KLC
+cross join rpt_pbh634.v_current_calendar cal
+GROUP BY KLC.GIFT_CLUB_ID_NUMBER),
+
+--- Count of KLC Years, but in last 5
+
+KLC5 AS (Select distinct KLC.GIFT_CLUB_ID_NUMBER,
+Count (distinct Club_END_DATE) as klc_fy_count_5
+from KLC
+cross join rpt_pbh634.v_current_calendar cal
+Where (KLC.Club_END_DATE = cal.CURR_FY - 1
+or KLC.Club_END_DATE = cal.CURR_FY - 2
+or KLC.Club_END_DATE = cal.CURR_FY - 3 
+or KLC.Club_END_DATE = cal.CURR_FY - 4
+or KLC.Club_END_DATE = cal.CURR_FY - 5)
+Group By KLC.GIFT_CLUB_ID_NUMBER),
+
+
+KLC_Final As (Select distinct KLC.GIFT_CLUB_ID_NUMBER,
+--- KLC FY Donor This Year
+KLC_Give_Ind.KSM_donor_cfy,
+--- KLC FY Donor Last Year 
+KLC_Give_Ind.KSM_donor_pfy1,
+--- KLC Total Years on Record 
+KLC_Count.klc_fy_count,
+--- KLC Total Years in the Last 5
+KLC5.klc_fy_count_5
+from KLC
+left join KLC_Count on KLC_Count.GIFT_CLUB_ID_NUMBER = KLC.GIFT_CLUB_ID_NUMBER
+left join KLC5 ON KLC5.GIFT_CLUB_ID_NUMBER = KLC.GIFT_CLUB_ID_NUMBER 
+left join KLC_Give_Ind on KLC_Give_Ind.GIFT_CLUB_ID_NUMBER = KLC.GIFT_CLUB_ID_NUMBER
+cross join rpt_pbh634.v_current_calendar cal)
+
+/* Final Query */
+
+Select distinct degrees.ID_NUMBER
+--- FY of most recent gift, pledge, or payment, (numeric)
+,Give.last_gift_fy --use FY function, give FY not date
+--- KSM Donor this Year
+,Donor_Ind.Donor_fy_ind as KSM_Donor_CFY
+--- KSM Donor last Year
+,Donor_Ind.Donor_pfy_ind as KSM_Donor_Pfy
+--- KSM Total Years of Giving (Numeric) 
+,G.fy_count as KSM_total_years_giving
+--- Current KSM AF donor status, (Donor, LYBUNT, PYBUNT, Lapsed, Non)
+,Give.af_status
+--- KLC FY Donor This Year
+,KLC_Final.KSM_donor_cfy as KSM_KLC_Donor_CFY
+--- KLC FY Donor Last Year 
+,KLC_Final.KSM_donor_pfy1 as KSM_KLC_Donor_PFY
+--- KLC Total Years on Record 
+,KLC_Final.klc_fy_count as KSM_KLC_Total_FY_Count
+--- KLC Total Years in the Last 5
+,KLC_Final.klc_fy_count_5 as KSM_KLC_Last5_PFY_Count
+FROM degrees
+Left Join a on a.id_number = degrees.ID_NUMBER
+Left Join give on give.id_number = degrees.ID_NUMBER
+Left Join g on g.id_number = degrees.ID_NUMBER
+Left Join KLC_Final on KLC_Final.GIFT_CLUB_ID_NUMBER = degrees.ID_NUMBER
+Left Join Donor_Ind on Donor_Ind.id_number = degrees.id_number
+where a.ANONYMOUS_DONOR is null
```

Comment:
Let's also redact anyone whose spouse is null. Easiest way would just be to join on spouse_id_number e.g.
Left Join a a_spouse On a_spouse.spouse_id_number = degrees.id_number
and also add another condition
where a_spouse.anonymous_donor is null